### PR TITLE
Refactor/binning

### PR DESCRIPTION
**Pull request by kayibal**
Changes necessary to make distributed binning algorithm work

### CODE REVIEW

File: sparsity/dask/core.py [hunk 1]
```diff
@@ -36,19 +36,23 @@ def _make_meta(inp):
...
 def optimize(dsk, keys, **kwargs):
     dsk, _ = cull(dsk, keys)
     return dsk
 
+
 def finalize(results):
     results = [r for r in results if not r.empty]
     return sp.SparseFrame.vstack(results)
 
+
```
Comment (michcio1234) [inline sparsity/dask/core.py]: WHO ARE YOU AND WHAT DID YOU DO WITH ALAN?!

File: sparsity/test/test_sparse_frame.py [hunk 2]
```diff
@@ -705,12 +705,10 @@ def test_repr():
     assert isinstance(res, str)
     assert '10x10000' in res
     assert '0 stored' in res
-    assert len(res.splitlines()) == 1 + 5 + 2
```
Comment (michcio1234) [inline sparsity/test/test_sparse_frame.py]: Why did the lines number change? I can't see any change that you made that should affect it...

Comment (michcio1234) [reply sparsity/test/test_sparse_frame.py]: Well, I hope our `__repr__` still makes sense then...

File: sparsity/dask/core.py (hunk 1 above)
Comment (kayibal) [reply sparsity/dask/core.py]: Hahaha

File: sparsity/test/test_sparse_frame.py (hunk 2 above)
Comment (kayibal) [reply sparsity/test/test_sparse_frame.py]: It should just lines change I think depending on your terminal size? Don't know...

File: .circleci/config.yml [hunk 3]
```diff
@@ -6,6 +6,7 @@ jobs:
       - image: drtools/dask:latest
     steps:
       - checkout
-      - run: pip install pytest pytest-cov moto dask==0.19.2 .
+      - run: pip install boto3==1.7.84 botocore==1.10.84 moto==1.3.6
+      - run: pip install pytest pytest-cov dask==1.0.0 .
```
Comment (michcio1234) [inline .circleci/config.yml]: Shouldn't this be consistent with setup.py? Or shouldn't it be done by `pip install .[test]`?

Comment (kayibal) [reply .circleci/config.yml]: Thx it is resolved

**Pull request merged**